### DISCUSSION TITLE
How to initiate plugin from native <select> element?

### DISCUSSION THREAD

**Discussion by sa-si-dev** [sa-si-dev/virtual-select#45] [Q&A] [answered]
How to initiate plugin from native `<select>` element?

**Comment by sa-si-dev** [comment]
[**UPDATE**] [v1.0.22](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.22) - This feature is supported in the plugin itself ([more details](https://sa-si-dev.github.io/virtual-select/#/?id=initialize-from-native-select-element-not-recommended))



**_--- NOT RECOMMENDED ---_**

It is not recommended to use native elements, to get the option's value and label to initiate the plugin.

If there is any rare scenario, where you could initiate plugin only from native select, use the below helper function in your project to initiate plugin.


```js
function initVirtualSelectFromNativeSelect(id) {
  var $ele = document.querySelector('#' + id);
  var $options = $ele.querySelectorAll('option');
  var options = [];

  /** getting options */
  $options.forEach(function ($option) {
    options.push({
      label: $option.innerHTML,
      value: $option.value,
    });
  });

  /** creating div element to initiate plugin and removing native element */
  var $newEle = document.createElement('div');
  $newEle.id = id;

  $ele.parentNode.insertBefore($newEle, $ele.nextSibling);
  $ele.remove();

  /** initiating virtual select */
  VirtualSelect.init({
    ele: '#' + id,
    options: options,
  });
}

```

**Reply by bes-internal** [reply]
Why is this not recommended? Can you explain what difficulties you faced during development and why you made such a decision?

**Reply by sa-si-dev** [reply]
> Why is this not recommended? Can you explain what difficulties you faced during development and why you made such a decision?

@bes-internal there are no difficulties in this method. The difference is the time taken to initiate the dropdown.

When I tried with 100000 options, initiating dropdown from native select took ~200ms whereas normal initiation took only ~40ms.

**Comment by eugenefoxx** [comment]
Is the required parameter supported?

**Reply by sa-si-dev** [reply]
> Is the required parameter supported?

As of now, no support for required attribute. Will implement it in the future release.

**Comment by sa-si-dev** [comment, marked as answer]
This feature is supported in the plugin itself  [v1.0.22](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.22)  ([more details](https://sa-si-dev.github.io/virtual-select/#/?id=initialize-from-native-select-element-not-recommended))
